src/__tests__/consumer/node_redis.spec.js

```diff
@@ -146,6 +148,242 @@ tape('NodeJS Redis', function (t) {
+
+        assert.equal(typeof client.track('nicolas@split.io', 'user', 'test.redis.event', 18).then, 'function', 'Track calls should always return a promise on Redis mode.');
+        assert.equal(typeof client.track().then, 'function', 'Track calls should always return a promise on Redis mode, even when parameters are incorrect.');
+
+        assert.true(await client.track('nicolas@split.io', 'user', 'test.redis.event', 18), 'If the event was succesfully queued the promise will resolve to true');
+        assert.false(await client.track(), 'If the event was NOT succesfully queued the promise will resolve to false');
+
+        await client.ready(); // promise already resolved
+        await client.destroy();
+        
+        exec(`echo "HGETALL ${config.storage.prefix}.SPLITIO.impressions.count" | redis-cli  -p ${redisPort}`, async (error, stdout) => {
+
+          const trackedImpressionCounts = stdout.split('\n').filter(line => line !== '');
+          assert.deepEqual(trackedImpressionCounts, expectedImpressionCount, 'Tracked impression counts should be stored in Redis');
+
+          // Validate stored impressions and events
+          exec(`echo "LLEN ${config.storage.prefix}.SPLITIO.impressions \n LLEN ${config.storage.prefix}.SPLITIO.events" | redis-cli  -p ${redisPort}`, (error, stdout) => {
+            if (error) assert.fail('Redis server should be reachable');
+
+            const trackedImpressionsAndEvents = stdout.split('\n').filter(line => line !== '').map(line => parseInt(line));
+            assert.deepEqual(trackedImpressionsAndEvents, [13, 2], 'Tracked impressions and events should be stored in Redis');
+
+            // Validate stored telemetry
+            exec(`echo "HLEN ${config.storage.prefix}.SPLITIO.telemetry.latencies \n HLEN ${config.storage.prefix}.SPLITIO.telemetry.exceptions \n HGET ${config.storage.prefix}.SPLITIO.telemetry.init nodejs-${version}/${HOSTNAME_VALUE}/${IP_VALUE}" | redis-cli  -p ${redisPort}`, (error, stdout) => {
+              if (error) assert.fail('Redis server should be reachable');
+
+              const [latencies, exceptions, configValue] = stdout.split('\n').filter(line => line !== '').map(JSON.parse);
+
+              assert.true(latencies > 0, 'There are stored latencies');
+              assert.true(exceptions === 0, 'There aren\'t stored exceptions');
+              assert.deepEqual(configValue, { oM: 1, st: 'redis', aF: 1, rF: 0 }, 'There is stored telemetry config');
+
+              // close server connection
+              server.close().then(assert.end);
+            });
+          });
+        });
+      });
+  });
+  
+  t.test('Regular usage - NONE strategy', assert => {
+    config.sync.impressionsMode = NONE;
+    initializeRedisServer()
+      .then(async (server) => {
+        const expectedConfig = '{"color":"brown"}';
+        const timeFrame = Date.now();
+        const expectedImpressionCount = [
```

Comment:
Nice work Emma!
Just a single comment.

If `expectedImpressionCount` is the same for the NONE and the OPTIMIZED tests, we can move it to line 71 and reuse in both tests. Same with `expectedUniqueKeys` and `expectedConfig` (this one is used 3 times)